subsys/bluetooth/mesh/friend.c

```diff
@@ -1265,9 +1261,18 @@ static void friend_timeout(struct k_work *work)
 	frnd->queue_size--;
 
 send_last:
+	buf = bt_mesh_adv_create(BT_MESH_ADV_DATA, FRIEND_XMIT, K_NO_WAIT);
+	if (!buf) {
+		BT_ERR("Unable to allocate friend adv buffer");
+		return;
+	}
+
+	net_buf_add_mem(buf, frnd->last->data, frnd->last->len);
```

Comment:
If I don't want to make a deep copy of the value, I can use `net_buf_simple_clone ` here? adv buf pool and friend buf pool are both `NET_BUF_POOL_FIXED ` so its `data_unref `is empty. Although it can run, does it destroy the encapsulation of the code?

Reply:
I guess you're talking about `user_data`, net_buf_simple_clone  doesn't involve this at all.

In general, I don't recommend this because it destroys the encapsulation of the code.

Reply:
No, I mean, we can't clone the buffer, because that would make the advertiser's buffer point to the friend buffer, which doesn't have the advertising header built-in, as far as I can tell. It would also break encapsulation, yeah, so I think we should just accept this copy, at least for this PR

Reply:
Agree.